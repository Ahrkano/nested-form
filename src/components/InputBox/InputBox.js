import React, { Component } from 'react';
import { returnInputBoxLogic } from './services/services';

import './InputBox.css';

class InputBox extends Component {
    render() {
        let input = returnInputBoxLogic(this.props);
        const marginLeft = 40 * this.props.level + 'px';

        return (
            <div style={{ marginLeft: marginLeft }} className="InputBox">
                <h4 className="InputBox__question">{this.props.question}</h4>
                {input}
            </div>
        );
    }
}

export default InputBox;
