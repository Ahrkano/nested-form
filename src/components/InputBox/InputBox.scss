@import 'shared/_colors.scss';

.InputBox {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    justify-content: center;
    background-color: $color-box-background;
    border: 1px solid $color-secondary;
    border-radius: 5px;
    box-shadow: 3px 3px 3px rgba(#000, 0.1);
    width: 300px;
    height: 100px;
    margin: 20px;

    &__question {
        display: block;
        margin: 0;
        margin-left: 20px;
        margin-bottom: 10px;

        color: hsl(0, 0%, 40%);

        font-size: 16px;
        font-weight: normal;
    }

    & p {
        display: inline-block;
        width: 100%;
        text-align: center;
        font-size: 10px;
        color: gray;
    }

    &__input {
        margin: 0 auto;
        padding: 3px;
        border-radius: 3px;
    }

    &__radio-wrapper {
        display: flex;
        width: 60%;
        justify-content: space-between;
        margin: 0 auto;
    }

    &__radio-group {
        cursor: pointer;
        padding: 2px 10px;
        // border: 1px dashed $color-primary;
        border-radius: 5px;

        background-color: rgba($color-primary, 0.2);

        &--yes {
            background-color: rgba($color-success, 0.2);
        }

        &--no {
            background-color: rgba($color-warning, 0.2);
        }

        & input,
        & label {
            cursor: pointer;
        }
    }
}
