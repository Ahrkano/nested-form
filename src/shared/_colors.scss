$color-primary: #0089BA;
$color-secondary: #D5CABD;
$color-warning: rgba(#d9534f, .8);
$color-warning-hover: #de716e;
$color-success: #95d095;
$color-success-hover: #5cb85c;