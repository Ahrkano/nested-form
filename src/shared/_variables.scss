$navbar-height: 50px;
$navbar-width: 500px;