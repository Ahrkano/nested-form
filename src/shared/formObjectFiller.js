export const formObjectFiller = function(questionArray) {
    if (questionArray) {
        const formObject = {}
        // populating formObject with questionObjects
        questionArray.forEach(questionItem => {
            formObject[questionItem.id] = {
                conditionType: questionItem.condition === 'noCondition' ? 'root' : 'conditional',
                parentId: questionItem.parentId,
                question: questionItem.question,
                inputType: questionItem.type,
                answer: '',
                level: questionItem.anchorLevel,
                conditionalQuestions: {}
            }; 
        });

        questionArray.forEach(questionItem => {

            if (questionItem.condition !== 'noCondition') {
                formObject[questionItem.parentId].conditionalQuestions[questionItem.id] = {
                    type: questionItem.condition,
                    value: questionItem.conditionValue
                };
            }
        });

        return formObject;
    }
}

export const rootQuestionsOrderArray = function(questionArray) {
    if (questionArray) {
        const rootQuestionsOrder = [];

        questionArray.forEach(questionItem => {
            if(questionItem.parentId === 'rootNode') {
                rootQuestionsOrder.push(questionItem.id);
            }
        });

        return rootQuestionsOrder;
    }
}