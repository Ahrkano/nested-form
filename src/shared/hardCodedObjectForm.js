export const objectForm = {
    question_C1: {
        answer: "",
        conditionType: "conditional",
        conditionalQuestions: {},
        inputType: "yesNo",
        parentId: "question_C0",
        question: "Third's question subinput"
    },
    question_A0: {
        answer: "",
        conditionType: "root",
        conditionalQuestions: {
            question_A1: {type: "equals", value: "yes"}
        },
        inputType: "yesNo",
        parentId: "rootNode",
        question: "First root question: choose yes",
    },
    question_A2: {
        answer: "",
        conditionType: "conditional",
        conditionalQuestions: {},
        inputType: "yesNo",
        parentId: "question_A1",
        question: "Third question",
    },
    question_A1: {
        answer: "",
        conditionType: "conditional",
        conditionalQuestions: {
            question_A2: {type: "equals", value: "second"}
        },
        inputType: "text",
        parentId: "question_A0",
        question: "Second question: input 'second'"
    },
    question_C0: {
        answer: "",
        conditionType: "root",
        conditionalQuestions: {
            question_C1: {type: "equals", value: "4"}
        },
        inputType: "number",
        parentId: "rootNode",
        question: "Third root question: input '4'"
    },
    question_B0: {
        answer: "",
        conditionType: "root",
        conditionalQuestions: {},
        inputType: "yesNo",
        parentId: "rootNode",
        question: "Second root question",
    }
}

export const rootQuestionsOrder = ["question_A0", "question_B0", "question_C0"];