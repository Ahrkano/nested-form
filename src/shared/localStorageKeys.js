export const MAIN_KEY = 'nestedFormData##';
