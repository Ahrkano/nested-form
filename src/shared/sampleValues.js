export const formObjectInstance = {
    question_8cbc0838: {
        conditionType: 'root',
        condition: 'noCondition',
        conditionValue: 'noCondition',
        parentId: 'rootNode',
        parentType: 'root',
        question: 'Write good',
        inputType: 'text',
        answer: '',
        level: 1,
        conditionalQuestions: {
            question_9a7475a1: {
                type: 'equals',
                value: 'good'
            }
        }
    },
    question_9a7475a1: {
        conditionType: 'conditional',
        condition: 'equals',
        conditionValue: 'good',
        parentId: 'question_8cbc0838',
        parentType: 'text',
        question: 'Click yes',
        inputType: 'yesNo',
        answer: '',
        level: 2,
        conditionalQuestions: {}
    },
    question_3ad9522f: {
        conditionType: 'root',
        condition: 'noCondition',
        conditionValue: 'noCondition',
        parentId: 'rootNode',
        parentType: 'root',
        question: 'Some question',
        inputType: 'yesNo',
        answer: '',
        level: 1,
        conditionalQuestions: {}
    }
};

export const rootQuestionOrderInstance = ['question_8cbc0838', 'question_3ad9522f'];
