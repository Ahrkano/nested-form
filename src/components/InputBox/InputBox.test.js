import React from 'react';
import renderer from 'react-test-renderer';
import InputBox from './InputBox';
import { formObjectInstance, rootQuestionOrderInstance } from '../../shared/sampleValues';

const questionId = rootQuestionOrderInstance[0];

describe('InputBox snapshot', () => {
    test('InputBox snapshot test', () => {
        const component = renderer.create(
            <InputBox
                key={questionId}
                id={questionId}
                question={formObjectInstance[questionId].question}
                inputType={formObjectInstance[questionId].inputType}
                value={formObjectInstance[questionId].answer}
                level={formObjectInstance[questionId].level}
                onInputChange={() => {}}
            />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
