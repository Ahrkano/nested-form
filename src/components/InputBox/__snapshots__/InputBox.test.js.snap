// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputBox snapshot InputBox snapshot test 1`] = `
<div
  className="InputBox"
  style={
    Object {
      "marginLeft": "40px",
    }
  }
>
  <h4
    className="InputBox__question"
  >
    Write good
  </h4>
  <input
    className="InputBox__input"
    onChange={[Function]}
    type="text"
    value=""
  />
</div>
`;
